src/App.tsx

```diff
@@ -1,56 +1,82 @@
-import React from 'react';
+import React, { useMemo, useState, useCallback } from 'react';
 import './App.scss';
+import classNames from 'classnames';
 import { peopleFromServer } from './data/people';
+import { PeopleList } from './components/PeopleList';
+import { Person } from './types/Person';
+
+function debounce(callback: (...args: string[]) => void, delay: number) {
+  let timerId = 0;
+
+  return (...args: string[]) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [person, setPerson] = useState<Person>();
+  const [query, setQuery] = useState('');
+  const [focus, setFocus] = useState(false);
+
+  const applyQuery = useCallback(debounce(setQuery, 1000), []);
+
+  const filteredPerson = useMemo(() => {
+    return peopleFromServer
+      .filter(human => human.name.toLowerCase().includes(query.toLowerCase()));
+  }, [query]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const handleBlur = () => {
+    setTimeout(() => {
+      setFocus(false);
+    }, 200);
+  };
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {!person && (
+          'No selected person'
+        )}
+        {person && (
+          `${person.name} (${person.born} - ${person.died})`
```

Comment:
```suggestion
        {!person && (
          'No selected person'
        )}
        {person && (
          `${person.name} (${person.born} - ${person.died})`
```
Use ternary operator instead
https://react.dev/learn/conditional-rendering#conditional-ternary-operator--